### DISCUSSION TITLE
String.Empty returns Some

### DISCUSSION THREAD

**Discussion by tjidde-nl** [louthy/language-ext#1186] [Q&A] [answered]
Hi,

I'm wondering, since I cant find it on the webs if this is intended or not.

If I have a piece of code where I set a String.Empty for example. If in my example the moon isn't blue it should return String.Empty. 

`var result = String.Empty
if(MoonIsBlue){
  result = GetMoonInfo();
}
return result;
`
I would expect that as being NONE since it is an empty object.  But the result I get is Some().

The reason I find this as unintended is that in your own docs you state NULL is the worst thing ever made in computer science.

I can't find the recommended way of handling this. 

(I know in the code I made above I could be if == true return GetMoonInfo(); But there are use cases where during logic it is possible to set a string as empty.) Or you start with an empty string and do some logic that might not give value to the string with certain conditions.

**Comment by CK-LinoPro** [comment, marked as answer]
`null` and `string.Empty` are two distinctly different values.
`Optional((string)null)` will return `None`, while `Optional(string.Empty)` will return `Some(string.Empty)`.
This is completely intentional since you might want to handle both cases separately.

If you want to filter for empty strings, you can do something like this:
```
string? someString;
var optionalString = Optional(someString).Filter(s => s != string.Empty);
```

**Reply by louthy** [reply]
This is the correct answer, but you can also use the built-in: `notEmpty`
```c#
var optionalString = Optional(someString).Filter(notEmpty);
```
Which will ignore white-space too.

Also:

> "you state NULL is the worst thing ever made in computer science."

I think that's probably regular-expressions.  `null` is a close second.
